RecoJets/JetProducers/plugins/MVAJetPuIdProducer.cc

```diff
@@ -0,0 +1,325 @@
+  desc.add<std::string>("jec", "AK4PF");
+  desc.add<bool>("residualsFromTxt", false);
+  desc.add<bool>("applyJec", false);
+  desc.add<edm::InputTag>("jetids", edm::InputTag(""));
+  desc.add<edm::InputTag>("rho", edm::InputTag("hltFixedGridRhoFastjetAll"));
+  desc.add<edm::InputTag>("jets", edm::InputTag("hltAK4PFJetsCorrected"));
+    edm::ParameterSetDescription vpsd1;
+    vpsd1.add<std::vector<std::string>>("tmvaVariables", {
+      "rho",
+      "nTot",
+      "nCh",
+      "axisMajor",
+      "axisMinor",
+      "fRing0",
+      "fRing1",
+      "fRing2",
+      "fRing3",
+      "ptD",
+      "beta",
+      "betaStar",
+      "DR_weighted",
+      "pull",
+      "jetR",
+      "jetRchg",
+    });
+    vpsd1.add<std::string>("tmvaMethod", "JetID");
+    vpsd1.add<bool>("cutBased", false);
+    vpsd1.add<std::string>("tmvaWeights", "RecoJets/JetProducers/data/MVAJetPuID.weights.xml.gz");
+    vpsd1.add<std::vector<std::string>>("tmvaSpectators", {
+      "jetEta",
+      "jetPt",
+    });
+    vpsd1.add<std::string>("label", "CATEv0");
+    vpsd1.add<int>("version", -1);
+    {
+      edm::ParameterSetDescription psd0;
+      psd0.add<std::vector<double>>("Pt2030_Tight", {
+        0.73,
+        0.05,
+        -0.26,
+        -0.42,
+      });
+      psd0.add<std::vector<double>>("Pt2030_Loose", {
+        -0.63,
+        -0.6,
+        -0.55,
+        -0.45,
+      });
+      psd0.add<std::vector<double>>("Pt3050_Medium", {
+        0.1,
+        -0.36,
+        -0.54,
+        -0.54,
+      });
+      psd0.add<std::vector<double>>("Pt1020_Tight", {
+        -0.83,
+        -0.81,
+        -0.74,
+        -0.81,
+      });
+      psd0.add<std::vector<double>>("Pt2030_Medium", {
+        0.1,
+        -0.36,
+        -0.54,
+        -0.54,
+      });
+      psd0.add<std::vector<double>>("Pt010_Tight", {
+        -0.83,
+        -0.81,
+        -0.74,
+        -0.81,
+      });
+      psd0.add<std::vector<double>>("Pt1020_Loose", {
+        -0.95,
+        -0.96,
+        -0.94,
+        -0.95,
+      });
+      psd0.add<std::vector<double>>("Pt010_Medium", {
+        -0.83,
+        -0.92,
+        -0.9,
+        -0.92,
+      });
+      psd0.add<std::vector<double>>("Pt1020_Medium", {
+        -0.83,
+        -0.92,
+        -0.9,
+        -0.92,
+      });
+      psd0.add<std::vector<double>>("Pt010_Loose", {
+        -0.95,
+        -0.96,
+        -0.94,
+        -0.95,
+      });
+      psd0.add<std::vector<double>>("Pt3050_Loose", {
+        -0.63,
+        -0.6,
+        -0.55,
+        -0.45,
+      });
+      psd0.add<std::vector<double>>("Pt3050_Tight", {
+        0.73,
+        0.05,
+        -0.26,
+        -0.42,
+      });
+      vpsd1.add<edm::ParameterSetDescription>("JetIdParams", psd0);
+    }
+    vpsd1.add<double>("impactParTkThreshold", 1.0);
+    std::vector<edm::ParameterSet> temp1;
+    temp1.reserve(1);
+
+    desc.addVPSet("algos", vpsd1, temp1);
+
+  descriptions.add("MVAJetPuIdProducer", desc);
+
+ }
+ 
+/*
```

Comment:
commented out code should be removed, unless there is a really good reason to keep